prac_07/programming_language.py

```diff
@@ -1,6 +1,6 @@
 """
 CP1404/CP5632 Practical - Suggested Solution
-Programming Language class with tests.
+Programming Language class with tests..
 """
 
 
```

Comment:
```
  def __init__(self, name, typing, reflection, year, arithmetic):
```

I would consider setting default values as part of best practice.  Something simple like this:

```
  def __init__(self, name="", typing="", reflection=False, year=0, arithmetic=False):
```